src/main/java/org/madhuri/app/message/dao/MessageDAO.java

```diff
@@ -87,5 +87,98 @@ public Message addMessage(Message msg) {
 
         return msg;
     }
-    
+
+	public Message getMessageById(long messageId) {
+		
+		Message message = null;
+	
+		try {
+            // 1. Get SQL connection
+            DbConnection dbConnection = new DbConnection();
+            Connection connection = dbConnection.getConnection();
+
+            // 2. Query the database for the message with the given messageId
+            String sql = "SELECT * FROM Message WHERE id=?";
+            PreparedStatement statement = connection.prepareStatement(sql);
+            statement.setLong(1, messageId);
+            ResultSet resultSet = statement.executeQuery();
+
+            // 3. If the message with the given messageId is found, create the Message object
+            while(resultSet.next()) {
+                long id = resultSet.getLong("id");
+                String username = resultSet.getString("username");
+                String messageContent = resultSet.getString("messageContent");
+                Timestamp timestamp = resultSet.getTimestamp("created");
+
+                // Convert the Timestamp to a Date object
+                Date created = new Date(timestamp.getTime());
+                String recipient = resultSet.getString("recipient");
+
+                // Create the Message object
+                message = new Message(id, messageContent, created, username, recipient);
```

Comment:
This logic is common in both the functions, can we write a separate function and reuse in both the places..